starlette/datastructures.py

```diff
@@ -4,6 +4,9 @@
 from shlex import shlex
 from urllib.parse import SplitResult, parse_qsl, urlencode, urlsplit
 
+if typing.TYPE_CHECKING:
+    from _typeshed import SupportsKeysAndGetItem
+
 from starlette.concurrency import run_in_threadpool
 from starlette.types import Scope
```

Comment:
Nitpick but should this be below the other imports?

```suggestion

from starlette.concurrency import run_in_threadpool
from starlette.types import Scope

if typing.TYPE_CHECKING:
    from _typeshed import SupportsKeysAndGetItem
```